pkg/rpc/client/client.go

```diff
@@ -40,6 +40,10 @@ type Client struct {
 	// cache is mostly filled in during Init(), but can also be updated
 	// during regular Client lifecycle.
 	cache cache
+
+	reqIDsLock  sync.RWMutex
+	reqIDs      map[int]struct{}
```

Comment:
And just drop this map? You're not likely to overflow uint64.

Reply:
NeoFS node is using a single Client over the whole node life. Are you sure that we can safely do this?

Also, this was mostly done for tests, because our server-side testing code for Client and WSClient is not so smart and `"id"=1` is hardcoded there. I've faced with this problem while refactoring `rpcClientErrorCases`.

Reply:
> NeoFS node uses a single Client over the whole node life

https://pkg.go.dev/time#Duration
> A Duration represents the elapsed time between two instants as an int64 nanosecond count. The representation limits the largest representable duration to approximately 290 years. 

You're not likely to perform a request once in a nanosecond. So even `int64` should be fine (maybe it's easier than `uint64` in some ways).

Reply:
Tests can be a bit more complicated though. We can leave some internal way to override default ID generation if we can't ensure symmetric response.

Reply:
Fixed, but I'd rather keep uint64 to avoid negative request IDs. Although jsonRPC specification doesn't tell anything about that, and even strings are allowed for ID.